lib/ProductOpener/Data.pm

```diff
@@ -56,6 +67,22 @@ use Action::Retry;
 my $client;
 my $action = Action::CircuitBreaker->new();
 
+=head1 FUNCTIONS
+
+=head2 execute_query()
+
+C<execute_query()> executes a query on the database.
+
+=head3 Arguments
+
+Takes in a subroutine with a return type T.
+
```

Comment:
I think it would be useful to add a full example of how this function is called.

e.g. in Display.pm, we call it like this:

```
eval {
		$cursor = execute_query(sub {
			return get_products_collection()->query($query_ref)->fields($fields_ref);
		});
	};
	if ($@) {
		$log->warn("MongoDB error", { error => $@ }) if $log->is_warn();
	}
	else {
		$log->info("MongoDB query ok", { error => $@ }) if $log->is_info();
	}
```

Reply:
I was thinking of doing just that. Will add a full example for this. Would it be okay if I add it to the `=head3 Arguments` or another head for such examples?

Reply:
@stephanegigandet I've added a synopsys section below. Please review

Reply:
The synopsys looks good, thank you